modules/ocf_backups/manifests/offsite.pp

```diff
@@ -19,5 +28,25 @@
     '/opt/share/backups/boto.cfg':
       source => 'puppet:///private/boto.cfg',
       mode   => '0600';
+
+    # Box.com credentials and API id/secret
+    '/opt/share/backups/box-creds.json':
+      source => 'puppet:///private/box-creds.json',
+      mode   => '0600';
+
+    # This changes each time a backup is made, so it is not stored on the
+    # puppetmaster and instead is just a local file on the backups server.
+    '/opt/share/backups/box-refresh-token':
+      mode => '0600';
```

Comment:
Yeah, I like the idea of doing a separate cronjob on the puppetmaster the most out of those. I agree that MySQL would be a bit heavy, but might be the easiest to update. The biggest issue I see with the separate cronjob idea is that it would have to run every hour (or slightly more often since every hour would create a small gap between valid tokens) to have a valid access token at all times just for the one time it needs it per week. That's probably fine, if a little wasteful, but better than having puppet runs fail when they shouldn't.